src/AstriaOracle.sol

```diff
@@ -36,21 +46,37 @@ contract AstriaOracle {
     // occurs when attempting to update the price for an uninitialized currency pair
     error UninitializedCurrencyPair(uint256 index);
 
+    // occurs when attempting to initialize an unauthorized currency pair, when authorization is required
+    error UnauthorizedCurrencyPair(bytes32 currencyPair);
+
     modifier onlyOracle() {
         require(msg.sender == ORACLE, "AstriaOracle: only oracle can update");
         _;
     }
 
-    constructor(address _oracle) {
+    constructor(address _oracle, bool _requireCurrencyPairAuthorization) Ownable(msg.sender) {
         ORACLE = _oracle;
+        requireCurrencyPairAuthorization = _requireCurrencyPairAuthorization;
+    }
+
+    function setRequireCurrencyPairAuthorization(bool _requireCurrencyPairAuthorization) external onlyOwner {
+        requireCurrencyPairAuthorization = _requireCurrencyPairAuthorization;
+    }
+
+    function authorizeCurrencyPair(bytes32 _currencyPair) external onlyOwner {
+        authorizedCurrencyPairs[_currencyPair] = true;
     }
 
     function initializeCurrencyPair(bytes32 _currencyPair, uint8 _decimals) external onlyOracle {
+        if (requireCurrencyPairAuthorization && !authorizedCurrencyPairs[_currencyPair]) {
+            revert UnauthorizedCurrencyPair(_currencyPair);
+        }
```

Comment:
Add to authorization list automatically for ability to enable authorization requirements in the future, disabled is implicit "authorize all new"

```suggestion
        }
        if (!requireCurrencyPairAuthorization && !authorizedCurrencyPairs[_currencyPair]) {
          authorizedCurrencyPairs[_currencyPair] = true;
        }
```

Reply:
Resolving in favor of another recommendation, realized that authorization only affects ability to initialize not ability to set price